tool/format_ignores.py

```diff
@@ -0,0 +1,226 @@
+
+    Returns
+    -------
+    tuple of (int, int)
+        (count of files that would be modified, total files checked)
+    """
+    modified_count = 0
+    total_count = 0
+
+    dir_path = Path(str(directory_path))
+
+    for ext in extensions:
+        for path in dir_path.rglob(f"*{ext}"):
+            total_count += 1
+            if _process_file(path, check_only=check_only):
+                modified_count += 1
+                if check_only:
+                    print(f"Would update {path}")
+
+    return modified_count, total_count
+
+
+def _parse_args(args: Sequence[str] | None = None, /) -> argparse.Namespace:
+    """
+    Parse command line arguments.
+
+    Parameters
+    ----------
+    args : sequence of str, optional
+        Command line arguments to parse. Defaults to sys.argv[1:].
+
+    Returns
+    -------
+    argparse.Namespace
+        Parsed arguments
+    """
+    parser = argparse.ArgumentParser(
+        description="Format pyright ignore comments in Python files.",
+    )
+    parser.add_argument(
+        "path",
+        nargs="?",
+        default=".",
+        help="Path to file or directory to process (default: current directory)",
+    )
+    parser.add_argument(
+        "--extensions",
+        default=".pyi",
+        help="Comma-separated list of file extensions to process (default: .pyi)",
+    )
+    parser.add_argument(
+        "--check",
+        action="store_true",
+        help="Check files without modifying them, exit with error if files "
+        "would be modified",
+    )
+
+    return parser.parse_args(args)
+
+
+def main(args: Sequence[str] | None = None, /) -> int:
+    """
+    Run the formatter.
+
+    Parameters
+    ----------
+    args : sequence of str, optional
+        Command line arguments to parse. Defaults to sys.argv[1:].
+
+    Returns
+    -------
+    int
+        Exit code: 0 for success, 1 for error or (in check mode) if files would
+        be modified
+    """
+    namespace = _parse_args(args)
+    extensions = tuple(ext.strip() for ext in namespace.extensions.split(","))
+
+    path = namespace.path
+    check_only = namespace.check
+    path_obj = Path(path)
+
+    if path_obj.is_dir():
+        modified, total = _process_directory(
+            path,
+            extensions=extensions,
+            check_only=check_only,
+        )
+        if check_only:
+            print(
+                f"Found {modified} files that would be modified out of {total} checked",
+            )
+            return 1 if modified > 0 else 0
+        print(f"Updated {modified} files out of {total} checked")
+        return 0
+    if path_obj.is_file():
+        modified = _process_file(path, check_only=check_only)
+        if check_only:
+            if modified:
+                print(f"File {path} would be modified")
+                return 1
+            print(f"File {path} is correctly formatted")
+            return 0
+        return 0
+    print(f"Path not found: {path}")
```

Comment:
usually error messages like these are printed to `stderr`, so with `print(..., file=sys.stderr)`